src/main/java/seedu/address/logic/commands/addcommand/AddMentorCommand.java

```diff
@@ -2,31 +2,54 @@
 
 import static java.util.Objects.requireNonNull;
 
+import seedu.address.commons.util.CollectionUtil;
 import seedu.address.logic.commands.CommandResult;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
 import seedu.address.model.entity.Mentor;
+import seedu.address.model.entity.Name;
 
 /**
  * Adds a {@link Mentor} to Alfred.
  */
 public class AddMentorCommand extends AddCommand {
 
-    /* Possible Fields: */
+    public static final String COMMAND_WORD = "addMentor";
+    private static final String MESSAGE_SUCCESS = "New mentor added: %s";
+    private static final String MESSAGE_DUPLICATE_MENTOR = "This mentor already exists in this Hackathon";
+
     private Mentor mentor;
+    private Name teamName;
 
     public AddMentorCommand(Mentor mentor) {
         requireNonNull(mentor);
         this.mentor = mentor;
     }
 
+    public AddMentorCommand(Mentor mentor, Name teamName) {
+        CollectionUtil.requireAllNonNull(mentor, teamName);
+        this.mentor = mentor;
+        this.teamName = teamName;
+    }
+
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        // See AddIssueCommand
+        if (this.teamName != null) {
+            // find team (or throw Exception)
+            // add Mentor to team
+            // return CommandResult
+        }
+
+        try {
+            model.addMentor(this.mentor);
+        } catch (Exception e) {
+            // Should I return new CommandResult(MESSAGE_DUPLICATE_MENTOR) instead?
```

Comment:
I was thinking instead of throwing an exception, i should just return a result saying it failed. Cuz I remember us saying Commands won't throw an exception

Reply:
I think command#execute method need to throw exception because the Ui catches this exception and prints out the exception message to the user :) It catches command exception and parse exception under MainWindow.java